firewood/src/v2/api.rs

```diff
@@ -64,6 +64,13 @@ pub enum Error {
     #[error("Incorrect root hash for commit: {provided:?} != {current:?}")]
     IncorrectRootHash { provided: HashKey, current: HashKey },
 
+    /// Invalid range
+    #[error("Invalid range: {first_key:?} > {last_key:?}")]
+    InvalidRange {
+        first_key: Vec<u8>,
+        last_key: Vec<u8>,
+    },
```

Comment:
I would rather have the key in the error message. Generally keys are not huge. I'm not sure I love having them rendered as hex bytes but that's better than not having them at all.

Why? Imagine you're the API user. Your app has been working for months. Suddenly you start getting this error, which you log, and it just says `first_key > last_key` or some such. You have no way of knowing what data you provided was wrong, unless you amend the error message with your parameters, which increases churn.

More about this in my aforementioned article.

Reply:
Yeah more leaning to @rkuris suggestion if the keys are not huge.

Reply:
Discussed with @richardpringle, he brought up this may be a bad practice, we're allocating unbounded keys just for an error message. That is a valid concern to me. @rkuris @danlaine

Reply:
Not sure why this got marked resolved @xinifinity. If you're worried about this, please open another issue for it. We could certainly only make the copy if the key is less than 1kb or something, but this isn't on the happy path and it's generally better to let the caller know what the bad data was.

Reply:
Created the issue to track https://github.com/ava-labs/firewood/issues/469